### DISCUSSION TITLE
RTSP Connection terminated

### DISCUSSION THREAD

**Discussion by DJ10177** [bluenviron/mediamtx#2116] [Questions]
I am publishing frames to the RTSP server using FFMPEG. My FFMPEG command is as follows: 

FILE* openPipeLine =  _popen("ffmpeg  -f rawvideo -r 10 -video_size 1280x720 -pixel_format bgr24 -i pipe: -vcodec libx264 -crf 25 -pix_fmt yuv420p -f rtsp rtsp://localIP:8554/mystream ", "wb"); The stream connects but is terminated after the time out period. The log message is as follows:
2023/07/28 13:28:01 INF [RTSP] [session 764e0840] is publishing to path 'mystream', with UDP, 1 track (H264)
2023/07/28 13:28:11 INF [RTSP] [conn 192.168.0.2:52995] closed (terminated)
2023/07/28 13:28:11 INF [RTSP] [session 764e0840] destroyed (session timed out)

It appears that the FFMPEG command does not send packets to the stream.  The ffmpeg command is in a MATLAB Mex function. However, what is strange is that if I create a stand alone Matlab executable application the stream does not disconnect. It only disconnects when running from matlab itself. 

Is there a way I can prevent this?

**Comment by aler9** [comment]
This doesn't look at all at something that depends on the server. Message "session timed out" means that the client is not sending frames. Make sure that the client is sending frames at a fixed interval.

[a comment by an automated account is not shown]
